go/runtime/host/helpers.go

```diff
@@ -0,0 +1,67 @@
+package host
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	"github.com/oasisprotocol/oasis-core/go/roothash/api/block"
+	"github.com/oasisprotocol/oasis-core/go/runtime/host/protocol"
+	"github.com/oasisprotocol/oasis-core/go/runtime/transaction"
+)
+
+var (
+	ErrInvalidArgument = errors.New("runtime: invalid argument")
+	ErrInternal        = errors.New("runtime: internal error")
+	ErrCheckTxFailed   = errors.New("runtime: check tx failed")
+)
+
+// RichRuntime provides higher-level functions for talking with a runtime.
+type RichRuntime interface {
+	Runtime
```

Comment:
what are the cases where we'll use Runtime instead of RichRuntime?

Reply:
I was thinking about that as well. Probably `RichRuntime` would mostly be used and it could just be merged into `Runtime`. But we also wouldn't want each `Runtime` implementation to implement all of these wrappers.  Maybe we could just provide a `BaseRuntime`?

Reply:
Okay yeah. I guess since we don't have plans to have a different implementation of CheckTx, all we have to gain that way is the code being more direct.

Reply:
feel free to split this out